### PR TITLE
Allow specific mangle options to be passed to gulp-uglify, fix incorrectly narrow type in uglify-js

### PR DESCRIPTION
**Pull request by Hexxeh**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: uglify-js accepts a number for max_line_len: https://github.com/mishoo/UglifyJS2/blob/b5ce1997112007d772e23bb04e09ed425bff608a/test/compress/max_line_len.js#L24, 
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Hexxeh** [comment]
Not sure what the best way to resolve this is, there appears to be a chain of typing dependencies. Is the general practice here to go add the version requirement to every package in the chain until it passes?

File: types/gulp-uglify/tsconfig.json [hunk 1]
```diff
@@ -14,12 +14,7 @@
         ],
         "types": [],
         "noEmit": true,
-        "forceConsistentCasingInFileNames": true,
-        "paths": {
```
Comment (alan-agius4) [inline types/gulp-uglify/tsconfig.json]: Why has this been removed?

File: types/uglify-js/index.d.ts [hunk 2]
```diff
@@ -165,7 +165,7 @@ export interface OutputOptions {
     indent_start?: boolean;
     inline_script?: boolean;
     keep_quoted_props?: boolean;
-    max_line_len?: boolean;
+    max_line_len?: boolean | number;
```
Comment (alan-agius4) [inline types/uglify-js/index.d.ts]: Please update this in `v2` as well

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Left some feedback

File: types/gulp-uglify/tsconfig.json (hunk 1 above)
Comment (Hexxeh) [reply types/gulp-uglify/tsconfig.json]: gulp-uglify uses uglify-js@3.x, so the mapping isn't needed so far as I can tell: https://github.com/terinjokes/gulp-uglify/blob/master/package.json#L14

Comment (alan-agius4) [reply types/gulp-uglify/tsconfig.json]: Afaik that is there to leave v2 maintained

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**